packages/flutter/lib/src/cupertino/date_picker.dart

```diff
@@ -479,86 +479,91 @@ class CupertinoDatePicker extends StatefulWidget {
-          if (longestText.length < hour.length) {
-            longestText = hour;
-          }
-        }
-      case _PickerColumnType.minute:
-        for (int i = 0; i < 60; i++) {
-          final String minute = localizations.datePickerMinute(i);
-          if (longestText.length < minute.length) {
-            longestText = minute;
-          }
-        }
-      case _PickerColumnType.dayPeriod:
-        longestText =
-          localizations.anteMeridiemAbbreviation.length > localizations.postMeridiemAbbreviation.length
-            ? localizations.anteMeridiemAbbreviation
-            : localizations.postMeridiemAbbreviation;
-      case _PickerColumnType.dayOfMonth:
-        int longestDayOfMonth = 1;
-        for (int i = 1; i <=31; i++) {
-          final String dayOfMonth = localizations.datePickerDayOfMonth(i);
-          if (longestText.length < dayOfMonth.length) {
-            longestText = dayOfMonth;
-            longestDayOfMonth = i;
-          }
-        }
-        if (showDayOfWeek) {
-          for (int wd = 1; wd < DateTime.daysPerWeek; wd++) {
-            final String dayOfMonth = localizations.datePickerDayOfMonth(longestDayOfMonth, wd);
-            if (longestText.length < dayOfMonth.length) {
-              longestText = dayOfMonth;
-            }
-          }
-        }
-      case _PickerColumnType.month:
-        for (int i = 1; i <= 12; i++) {
-          final String month = standaloneMonth
-              ? localizations.datePickerStandaloneMonth(i)
-              : localizations.datePickerMonth(i);
-          if (longestText.length < month.length) {
-            longestText = month;
-          }
-        }
-      case _PickerColumnType.year:
-        longestText = localizations.datePickerYear(2018);
-    }
-
-    assert(longestText != '', 'column type is not appropriate');
-
-    return TextPainter.computeMaxIntrinsicWidth(
+// Estimate the minimum width that each column needs to layout its content.
+static double _getColumnWidth(
+  _PickerColumnType columnType,
+  CupertinoLocalizations localizations,
+  BuildContext context,
+  bool showDayOfWeek, {
+  bool standaloneMonth = false,
+}) {
+  // Helper function to measure the width of a given text.
+  double measureTextWidth(String text) {
+    final TextPainter textPainter = TextPainter(
       text: TextSpan(
         style: _themeTextStyle(context),
-        text: longestText,
+        text: text,
       ),
       textDirection: Directionality.of(context),
-    );
+    )..layout();
+    return textPainter.size.width;
+  }
+
+  // Helper function to find the maximum width from a list of strings.
+  double maxTextWidth(List<String> texts) {
+    return texts.map(measureTextWidth).reduce((double a, double b) => a > b ? a : b);
+  }
+
+  switch (columnType) {
+    case _PickerColumnType.date:
+      final List<String> dates = <String>[
+        for (int i = 1; i <= 12; i++)
+          localizations.datePickerMediumDate(DateTime(2018, i, 25))
```

Comment:
You have to find the longest date + month combination I think? Here the method is supposed to return the width of `year + month + day` right?

Reply:
Really. But I adjusted the logic again, to avoid errors.